verify_rpms/rpm_verifier.py

```diff
@@ -44,85 +47,83 @@ def get_rpmdb(container_image: str, target_dir: Path, runner: Callable = run) ->
             str(rpmdb),
         ],
         capture_output=True,
         text=True,
         check=True,
     ).stdout.splitlines()
-    return [
-        rpm.split()[0]
-        for rpm in rpm_strs
-        if "Key ID" not in rpm and not rpm.startswith("gpg-pubkey")
-    ]
+    return rpm_strs
 
 
-def generate_output(processed_images: Iterable[ProcessedImage]) -> tuple[bool, str]:
+def get_unsigned_rpms(rpms: list[str]) -> list[str]:
     """
-    Generate the output that should be printed to the user based on the processed images
-    results.
-
-    :param processed_images: each element contains a name and a list of unsigned images
-
-    :returns: The status and summary of the image processing.
+    Get all unsigned RPMs from the list of RPMs
+    Filters out the `gpg-pubkey` and return the unsigned RPMs
+    :param rpms: list of RPMs
+    :return: list of unsigned RPMs
     """
-    failures = True
-    with_unsigned_rpms = [img for img in processed_images if img.unsigned_rpms]
-    with_error = [img for img in processed_images if img.error]
-    if not with_unsigned_rpms and not with_error:
-        output = "No unsigned RPMs found."
-        failures = False
-    else:
-        output = "\n".join([str(img) for img in with_unsigned_rpms])
-        output = f"Found unsigned RPMs:\n{output}" if with_unsigned_rpms else ""
-        error_str = "\n".join([f"{img.image}: {img.error}" for img in with_error])
-        output = f"{output}\nEncountered errors:\n{error_str}" if error_str else output
-    return failures, output.lstrip()
+    unsigned_rpms: list[str] = []
+    for rpm in rpms:
+        if "Key ID" not in rpm and not rpm.startswith("gpg-pubkey"):
+            unsigned_rpms.append(rpm.split()[0])
+    return unsigned_rpms
 
 
-def parse_image_input(image_input: str) -> list[str]:
+def generate_output(processed_image: ProcessedImage) -> tuple[bool, str]:
     """
-    Input is either an image reference or snapshot data in json format.
-    Try parsing as json and extract the images.
-    If failing to parse as json, assume it's an image reference as one-element list.
+    Generate the output that should be printed to the user based on the processed images
+    results.
+    :param processed_image: ProcessedImage that contains all the rpms information
+    :returns: The status and summary of the image processing.
     """
-    try:
-        snapshot = loads(s=image_input)
-    except JSONDecodeError:
-        return [image_input]
-    return [component["containerImage"] for component in snapshot["components"]]
+    failure = True
+
+    if not processed_image.unsigned_rpms and not processed_image.error:
+        output = f"No unsigned RPMs in {processed_image.image}"
+        failure = False
+    else:
+        output = (
+            f"Found unsigned RPMs in {processed_image.image}:\n{processed_image.unsigned_rpms}"
+            if processed_image.unsigned_rpms
+            else ""
+        )
+        output = (
```

Comment:
We're discarding the list of unsigned RPMs from the previous line. Before I believe we were collecting both. Is that the intention?

Reply:
We are outputting the list of `unsigned_rpms` only if its not empty

Reply:
Or in case of an error, we print the error we encountered

Reply:
Before, in case of unsigned rpms and an error, we were printing both. This is now dropped. Was that the intention?

Reply:
Yes, that was the intention.
Before we could have more than one image so there could be a situation where we got unsigned RPMs for one image and an error for the other, so we had to print them both.
Now we only get one image as an input, so it either has no unsigned RPMs (print `No unsigned RPMs...` , or has unsigned RPMS (printing the list of unsigned RPMS) or we got an error (print the error)

Reply:
got it